protocol.go

```diff
@@ -68,20 +68,16 @@ func ParseResponse(reader io.Reader, writer io.Writer) (*FileInfos, error) {
 				return nil, err
 			}
 
-			message, err = bufferedReader.ReadString('\n')
-			if err == io.EOF {
+			if bufferedReader.Buffered() == 0 {
```

Comment:
I am a bit lost here what this is supposed to mean. `Buffered` returns the number of bytes remaining in the read buffer, but as far as I see the buffer should already be empty since we read all its information for parsing the time message. We do not make any `Ack`s in between so we do not expect another message back from the remote.  My suggestion is to add more comments to this function to make it more clear why certain checks are necessary and what part of the protocol we are considering at each line in the code.

Reply:
https://github.com/charmbracelet/wish/blob/e6e9fc4c8a253cf263334efa95d31e7af7034970/scp/scp.go#L128

wish is a popular ssh custom server. Their implementation allows for sending out the time and permissions in the same message. There is an argument that it's not following the protocol, but being so popular, I thought it would be nice to accommodate and take into consideration the instances in which someone could make a custom ssh implementation.

If you believe we shouldn't accommodate that kind of custom implementation, then I will remove that piece of logic to keep it clean.

Reply:
I see, I think you make a good argument for keeping it, but make sure that it does not interfere with implementations that do follow the protocol.